### ISSUE TITLE
Intellij plugin failing on ide startup

### ISSUE THREAD

**Issue by greenhull** [antlr/intellij-plugin-v4#605]
```
Details: Current thread: Thread[Alarm Pool,4,main] 1349399959
Dispatch thread: false; isDispatchThread(): false
SystemEventQueueThread: Thread[AWT-EventQueue-0,6,main] 160545550

java.lang.Throwable: Read access is allowed from inside read-action (or EDT) only (see com.intellij.openapi.application.Application.runReadAction())
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:228)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1013)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.evaluateValidity(FileManagerImpl.java:596)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.evaluateValidity(FileManagerImpl.java:592)
	at com.intellij.psi.impl.source.PsiFileImpl.isValid(PsiFileImpl.java:163)
	at com.intellij.psi.impl.source.PsiFileImpl.getText(PsiFileImpl.java:300)
	at org.antlr.intellij.plugin.validation.GrammarIssuesCollector.collectGrammarIssues(GrammarIssuesCollector.java:36)
	at org.antlr.intellij.plugin.ANTLRv4ExternalAnnotator.doAnnotate(ANTLRv4ExternalAnnotator.java:37)
	at org.antlr.intellij.plugin.ANTLRv4ExternalAnnotator.doAnnotate(ANTLRv4ExternalAnnotator.java:24)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:209)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:203)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.runChangeAware(ExternalToolPass.java:278)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$1(ExternalToolPass.java:181)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:366)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:344)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:180)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:334)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:324)
	at com.intellij.util.ui.update.MergingUpdateQueue.doExecute(MergingUpdateQueue.java:281)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:264)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:238)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:241)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:377)
	at com.intellij.util.Alarm$Request.run(Alarm.java:366)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:272)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:243)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:216)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:212)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:205)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

-------------------------------------------------------------------------------------------------------------------------------
IDE version info
-------------------------------------------------------------------------------------------------------------------------------

IntelliJ IDEA 2022.3.2 RC (Ultimate Edition)
Build #IU-223.8617.20, built on January 16, 2023
Licensed to Chester John
Subscription is active until March 9, 2023.
Runtime version: 17.0.5+1-b653.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 36
Registry:
    editor.focus.mode.color.light=e0861f
    ide.editor.tab.selection.animation=true
    ide.new.editor.tabs.vertical.borders=true
    ide.balloon.shadow.size=0
    editor.focus.mode.color.dark=e0861f

Non-Bundled Plugins:
    com.chylex.intellij.inspectionlens (1.1.1)
    org.antlr.intellij.plugin (1.19.3-3)
    Gradle View (4.0.0)
    org.sylfra.idea.plugins.linessorter (1.0.1)
    String Manipulation (9.6.1)
    com.github.platan.gradle-dependencies-formatter (0.6.1)
    tarn.gradle.dependencies (1.6.2)
    MavenRunHelper (4.23.222.2964.0)
    GrepConsole (12.15.211.6693.1)
    com.chrisrm.idea.MaterialThemeUI (8.4.1)
    GradleDependenciesHelper (1.16)
    org.sonarlint.idea (7.3.0.59206)

Kotlin: 223-1.8.0-release-345-IJ8617.20

**Comment by anh977** [comment]
The Plugin worked until I upgraded to:
IntelliJ IDEA 2023.1 (Community Edition)
Build #IC-231.8109.175, built on March 28, 2023
Runtime version: 17.0.6+10-b829.5 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.15.0-67-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 1494M
Cores: 8
Registry:
    scala.erase.compiler.process.jdk.once=false

Non-Bundled Plugins:
    org.antlr.intellij.plugin (1.19.3-3)
    org.intellij.scala (2023.1.15)
    org.nik.presentation-assistant (1.0.10)

Kotlin: 231-1.8.20-IJ8109.175
Current Desktop: MATE
Now I get the same error everytime on startup, after I clear it, and when I use Test Rule (but it still shows parse trees)

**Comment by anh977** [comment]
A rollback to 2022.3.3 stopped the plugin error.

IntelliJ IDEA 2022.3.3 (Community Edition)
Build #IC-223.8836.41, built on March 9, 2023
Runtime version: 17.0.6+1-b653.34 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.15.0-67-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 1494M
Cores: 8
Registry:
    scala.erase.compiler.process.jdk.once=false

Non-Bundled Plugins:
    org.antlr.intellij.plugin (1.19.3-3)
    org.intellij.scala (2022.3.20)
    org.nik.presentation-assistant (1.0.10)

Kotlin: 223-1.8.0-release-345-IJ8836.41
Current Desktop: MATE

**Comment by Re-TinyLin** [comment]
The same issue  after upgraded to ntelliJ IDEA 2023.1

**Comment by jiangtao69039** [comment]
is resolved? the same issue when I upgrade to 2023.1
![image](https://user-images.githubusercontent.com/20572131/228742928-3f6e954f-c74d-44f2-9302-27d1716f10b2.png)

**Comment by bjansen** [comment]
It's fixed in 1.20, which I hope to release this week.
